src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/Path.java

```diff
@@ -123,8 +123,26 @@ public boolean pathsAreTheSame(final Path<T,E> path) {
     public boolean containsVertex(final T v) {
         Utils.nonNull(v, "Vertex cannot be null");
 
-        // TODO -- warning this is expensive.  Need to do vertex caching
-        return getVertices().contains(v);
+
+        if ( getEdges().isEmpty() ) {
+            return false;
+        } else {
+            boolean first = true;
+            for ( final E e : getEdges() ) {
+                if ( first ) {
```

Comment:
done, replaced all of this with 
```
  return v.equals(getFirstVertex()) || edgesInOrder.stream().map(graph::getEdgeTarget).anyMatch(v::equals);
```

which does not create a list but does create some objects to it may come up on profiling again (in which case we'll rewrite for speed)